tap-agent/src/agent/sender_account.rs

```diff
@@ -214,17 +287,32 @@ impl Actor for SenderAccount {
         );
         match message {
             SenderAccountMessage::UpdateReceiptFees(allocation_id, unaggregated_fees) => {
-                let tracker = &mut state.sender_fee_tracker;
-                tracker.update(allocation_id, unaggregated_fees.value);
+                state
+                    .sender_fee_tracker
+                    .update(allocation_id, unaggregated_fees.value);
 
-                if tracker.get_total_fee() >= state.config.tap.rav_request_trigger_value.into() {
+                // Eagerly deny the sender (if needed), before the RAV request. To be sure not to
+                // delay the denial because of the RAV request, which could take some time.
+                if !state.deny {
+                    state.maybe_deny().await;
+                }
```

Comment:
If this is already a `maybe_deny`, why do you have an `if` stmt.

If the reason is to not call an async function, something like the following makes more sense: 
```
if state.should_deny() {
    state.add_to_denylist().await;
}
```

Reply:
The reason was to not spam the DB if the sender is already denied. (and the opposite for the other one). Though I could simplify this further. I think I can simplify this now

Reply:
Yup, but you could add the `state.deny` inside the `maybe` function

Reply:
If I get a receipt that pushes me over the limit, but request a RAV instead of immediately denying, I'm exposing myself to more risk. The RAV request can take a lot of time during which I'll continue receiving queries.